core/src/main/java/io/onetable/delta/DeltaSchemaExtractor.java

```diff
@@ -146,7 +146,7 @@ public OneSchema toOneSchema(StructType structType) {
 
   private OneSchema toOneSchema(
       DataType dataType, String parentPath, boolean nullable, String comment) {
-    Map<OneSchema.MetadataKey, Object> metadata = null;
+    Map<OneSchema.MetadataKey, Object> metadata = new EnumMap<>(OneSchema.MetadataKey.class);
```

Comment:
This will create an EnumMap per field. EnumMap requires an array the size of the Enum values to be created when creating an instance. Leaving this as `null` will allow the optimization I previously made to avoid map allocation.